lib/pbench/server/api/resources/server_configuration.py

```diff
@@ -139,48 +144,80 @@ def _put(self, params: ApiParams, _) -> Response:
                         f"No value found for key system configuration key {key!r}",
                     )
 
-            if len(keydup) > 0:
-                raise APIAbort(
-                    HTTPStatus.BAD_REQUEST,
-                    f"Redundant parameters specified in the JSON request body: {sorted(keydup)!r}",
-                )
-
             try:
                 ServerConfig.set(key=key, value=value)
                 return jsonify({key: value})
             except ServerConfigBadValue as e:
-                raise APIAbort(HTTPStatus.BAD_REQUEST, str(e))
+                raise APIAbort(HTTPStatus.BAD_REQUEST, str(e)) from e
             except ServerConfigError as e:
-                raise APIAbort(HTTPStatus.INTERNAL_SERVER_ERROR, str(e))
+                raise APIAbort(HTTPStatus.INTERNAL_SERVER_ERROR, str(e)) from e
+
+    def put_body(self, params: ApiParams) -> Response:
```

Comment:
If you make a change for some other reason, consider moving the `return` statement at line 149 out of the `try` block:  this better parallels the code in `put_body()`, and, in the event that `jsonify()` raises an exception, it won't be covered by the `try` block (not that that makes anything but a philosophical difference at the moment, but...).